### ISSUE TITLE
Allow tracing of time.c

### ISSUE THREAD

**Issue by mpe** [linuxppc/linux#91]
Currently all of `arch/powerpc/kernel/time.c` is excluded from tracing because of:

```
# timers used by tracing
CFLAGS_REMOVE_time.o = -mno-sched-epilog $(CC_FLAGS_FTRACE)
```

However x86 doesn't do this, although it does exclude `tsc.c`. So we should dig into what (if anything) in our `time.c` can't be traced and split it out into a separate file.

**Comment by mpe** [comment]
Dropping the Makefile line (so allowing tracing) appears to work, at least we boot and pass all the ftrace selftests, and function graph trace works. Perhaps there's a corner case somewhere that doesn't work?

**Comment by rnav** [comment]
On 2017/03/17 03:19AM, Michael Ellerman wrote:
> Dropping the Makefile line (so allowing tracing) appears to work, at least we boot and pass all the ftrace selftests, and function graph trace works. Perhaps there's a corner case somewhere that doesn't work?

Hmm... weird. I'd expect sched_clock() to cause problems as it isn't 
notrace.  I'll take a look.

**Comment by rnav** [comment]
sched_clock() has the notrace annotation in linux/sched.h and that is 
ensuring that it doesn't get traced. It would be better to add that 
annotation in time.c as well and look at which other functions (if any) 
need notrace.

**Comment by santoshs** [comment]
Patch is at: http://patchwork.ozlabs.org/patch/778146/

**Comment by mpe** [comment]
Fixed in 6b847d795cf4ab3e574f4fcf7193fe, thanks @santoshs ! :cake:
